File: generators/java/src/main/java/dqualizer/dqlang/archive/k6configurationrunner/dqlang/options/Scenario.java
package dqualizer.dqlang.archive.k6configurationrunner.dqlang.options;

import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonTypeInfo;
import dqualizer.dqlang.archive.k6adapter.dqlang.k6.options.ConstantScenario;
import dqualizer.dqlang.archive.k6adapter.dqlang.k6.options.RampingScenario;

@JsonTypeInfo(
        use = JsonTypeInfo.Id.NAME,
        property = "executor")
@JsonSubTypes({
        @JsonSubTypes.Type(value = ConstantScenario.class, name = "constant-vus"),
        @JsonSubTypes.Type(value = RampingScenario.class, name = "ramping-vus")
})
public abstract class Scenario {
}

File: src/main/kotlin/dqualizer/dqlang/archive/k6configurationrunner/dqlang/options/Scenario.kt
package dqualizer.dqlang.archive.k6configurationrunner.dqlang.options

import com.fasterxml.jackson.annotation.JsonSubTypes
import com.fasterxml.jackson.annotation.JsonTypeInfo

@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "executor")
@JsonSubTypes(
    JsonSubTypes.Type(
        value = dqualizer.dqlang.archive.k6adapter.dqlang.k6.options.ConstantScenario::class,
        name = "constant-vus"
    ), JsonSubTypes.Type(
        value = dqualizer.dqlang.archive.k6adapter.dqlang.k6.options.RampingScenario::class,
        name = "ramping-vus"
    )
)
abstract class Scenario  
